python/cuml/cuml/internals/base.pyx

```diff
@@ -798,10 +798,6 @@ class UniversalBase(Base):
         if not hasattr(self, "_gpuaccel"):
             return cuml.global_settings.device_type
 
-        # if using accelerator and doing inference, always use GPU
-        elif func_name not in ['fit', 'fit_transform', 'fit_predict']:
-            device_type = DeviceType.device
-
```

Comment:
This particular change appears to introduce a regression, see https://github.com/rapidsai/cuml/pull/6554#issuecomment-2822299357 .

@dantegd can you comment on the initial motivation for this?

Reply:
Technically speaking we can run inference in many cases even if training was done on GPU, but I agree with the change done in this PR after analyzing the behavior that @jcrist mentions. The CPU to GPU transfer eats a lot of the time that the inference acceleration gains.